packages/internal/fetch-node/src/unicast.ts

```diff
@@ -115,25 +115,20 @@ export function unicastFetchWrap<C = FetchContext>({
             },
           })
 
-          const headers = new Headers(init?.headers)
-          headers.set('connection', 'close') // Proactively close the connection
-
           try {
-            return await fetch.call(this, input, {
+            const headers = new Headers(init?.headers)
+            headers.set('connection', 'close') // Proactively close the connection
+
+            const response = await fetch.call(this, input, {
               ...init,
               headers,
               // @ts-expect-error non-standard option
               dispatcher,
             })
-          } finally {
-            // Free resources (we cannot await here since the response was not
-            // consumed yet).
-            void dispatcher.close().catch((err) => {
-              // No biggie, but let's still log it
-              console.warn('Failed to close dispatcher', err)
-            })
 
             if (!didLookup) {
+              await response.body?.cancel()
```

Comment:
Do you know why in the undici examples they drain the body with `for await ...` rather than `cancel()`? https://undici.nodejs.org/#/?id=garbage-collection

Reply:
In the docs they do say to "consume or cancel" so I am sure this is fine, just a bit odd they don't use `cancel()` since it's so much more direct!

Reply:
consuming will keep the socket open (for future re-use). My guess is that `cancel()` causes the socket to be closed, which is what we actually want.